### PR TITLE
CSS: adjust media query breakpoint to improve tablet readability

### PR DESCRIPTION
**Pull request by nachoparker**
see https://github.com/nextcloud/news/pull/467

| <= 938 | > 938 |
|:---------:|:------:|
|dev skjnldsv com_apps_files__dir=_|dev skjnldsv com_apps_files__dir=_ (1)|

### CODE REVIEW

**Comment by jancborchardt** [comment]
What do you think @nextcloud/designers?

**Comment by skjnldsv** [comment]
Damn, I had a pending branch for that :D
Thanks a lot for taking care of that @nachoparker 
There are some javascript to edit as well!

https://github.com/nextcloud/server/commit/1a362809efc683b0204b1d60baede5eb6ab19d28

**Comment by nachoparker** [comment]
@skjnldsv thanks for pointing that out. I tested briefly with the main default apps and I didn't see that we needed any more changes, but since you have made more progress on this, should we merge your changes here? or do you have a standing PR?

In that case we can close this as duplicate

**Comment by skjnldsv** [comment]
@nachoparker I did not opened the pr yet :)
Please check my commit and adapt this pr accordingly, or I can also open one based on my branch :wink:

**Comment by nachoparker** [comment]
(sorry I deleted my remote branch accidentally)

@skjnldsv I looked at your changes and tweaked them a bit. The only big change is in `apps/files/css/mobile.css`, where I feel that it is better to simplify the media query. PTAL.

**Comment by skjnldsv** [comment]
Another one: https://github.com/nextcloud/server/blob/c91ba00e0e10130964c936ceb2db2f88caff5d92/apps/files_sharing/css/public.scss#L198 :)

**Comment by skjnldsv** [comment]
:trumpet:

**Comment by nachoparker** [comment]
let me get you an animation, it will be easier to explain

**Comment by nachoparker** [comment]
Ok, this is with the rule 

```js
@media only screen and (max-width: 1024px) and (min-width: 769px), only screen and (max-width: 688px) {
```

before

, and this is with 

```js
@media only screen and (max-width: $breakpoint-mobile) {
```

after

If we want to keep three cut points, there needs to be more adjustment to be made. I like it simple but that's just my preference.

**Comment by skjnldsv** [comment]
@jancborchardt :)

**Comment by jancborchardt** [comment]
@nachoparker looking at the gifs:
- The first one looks good, but when it’s narrower without the left navigation it only shows the right details inbetween, and not when wider? This version would be good if that’s fixed.
- The second one has a looong period of where it doesn’t show the left navigation but also does not show any details on the right. This is a bit strange → see what I said in point 1 :)

**Comment by nachoparker** [comment]
Thanks @jancborchardt the first one is what you already have, only substituting 938px -> 1024px, that's what I mean that it needs some more tweaking because it doesn't look good (and I proposed number 2).

Let's see if we can find a sweet spot

**Comment by nachoparker** [comment]
Ok, so if we want three breakpoints I think this is exactly what we are after -> https://github.com/nextcloud/server/pull/15199/commits/f902ee73f59442359d86f979cfead8fd505b7286

3bpoints

File: apps/files/css/mobile.scss [hunk 1]
```diff
@@ -1,4 +1,4 @@
-@media only screen and (max-width: $breakpoint-mobile) {
+@media only screen and (max-width: 938px) and (min-width: $breakpoint-mobile + 1), only screen and (max-width: 688px) {
```
Comment (skjnldsv) [inline apps/files/css/mobile.scss]: 868 was basically the 688+250 (nav width)
We might be able to find the use of the mobile breakpoint somewhere here :)

Comment (skjnldsv) [reply apps/files/css/mobile.scss]: I think the idea behind it is that we found that the min width the filelist should have was 688px, (with full view, modified, size, actions.
..) So those breakpoints were there for that.
@jancborchardt ?

**Comment by jancborchardt** [comment]
@nachoparker nice, that last gif you posted looks perfect to me! :+1: What do you think @nextcloud/designers?

**Comment by skjnldsv** [comment]
I pushed a commit to use the $navigation-width variable.
Peek 29-05-2019 06-23

**Review by skjnldsv — APPROVED** [review]

**Review by skjnldsv — APPROVED** [review]
rebased and ready to go

**Comment by jancborchardt** [comment]
@skjnldsv hmm, the gif by @nachoparker looks slightly better than the last one you posted, because it has 1 less breaking step:
1. Show navigation and file details on right
2. Hide navigation, still show file details
3. Hide navigation and file details (mobile view)

You have a strange step between 1 and 2 with "Show navigation but hide file details", which doesn’t seem necessary. The content should always take precedence over the navigation. That is, before any content is hidden, the navigation should be hidden before, as done in @nachoparker’s gif.

What do you think?

**Comment by skjnldsv** [comment]
> What do you think?

Sure :)
Then let's adjust the min width for the file table.
I just used what we were supposed to use before (original calculation).

But looking at the first gif, we don't need two breakpoint then. Only one in 688px.

**Review by jancborchardt — APPROVED** [review]
Looking very good now! :+1: 

(We should also look at how it works with the right sidebar, but that’s a follow-up. :)

**Comment by go2sh** [comment]
@skjnldsv Just a minor nitpick. You commited some git merge/rebase LOCAL REMOTE files to the repo. ;-)

**Comment by skjnldsv** [comment]
@go2sh what do you mean?

**Comment by nachoparker** [comment]
Thanks gents!

**Pull request merged**